extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/devconsole/HibernateSearchSupplier.java

```diff
@@ -21,24 +24,34 @@ public class HibernateSearchSupplier implements Supplier<List<DevUiIndexedEntity
     }
 
     @Override
-    public List<DevUiIndexedEntity> get() {
+    public Map<String, List<DevUiIndexedEntity>> get() {
         if (!isEnabled()) {
-            return Collections.emptyList();
+            return Collections.emptyMap();
         }
-        SearchMapping mapping = searchMapping();
-        if (mapping == null) {
-            return Collections.emptyList();
+        Map<String, SearchMapping> mappings = searchMapping();
+        if (mappings.isEmpty()) {
+            return Collections.emptyMap();
         }
-        return mapping.allIndexedEntities().stream().map(DevUiIndexedEntity::new).sorted()
-                .collect(Collectors.toList());
+        return mappings.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey,
+                entry -> entry.getValue().allIndexedEntities().stream().map(DevUiIndexedEntity::new).sorted()
+                        .collect(Collectors.toList())));
     }
 
     private boolean isEnabled() {
         return runtimeConfig.defaultPersistenceUnit.enabled;
     }
 
-    public static SearchMapping searchMapping() {
-        return Arc.container().instance(SearchMapping.class).get();
+    public static Map<String, SearchMapping> searchMapping() {
+        return Arc.container().beanManager().getBeans(SearchMapping.class)
+                .stream().map(bean -> (InjectableBean<SearchMapping>) bean)
+                .collect(Collectors.toMap(HibernateSearchSupplier::getPersistenceUnitName,
+                        bean -> Arc.container().instance(bean).get()));
+    }
+
+    private static String getPersistenceUnitName(InjectableBean<SearchMapping> bean) {
+        return bean.getQualifiers().stream()
+                .filter(annotation -> annotation.annotationType().isAssignableFrom(PersistenceUnit.class))
+                .map(PersistenceUnit.class::cast).map(PersistenceUnit::value).findFirst().get();
```

Comment:
Alternatively, to get the persistence unit name from a `SearchMapping` instance directly, you can call `searchMapping.toOrmSessionFactory().unwrap(SessionFactoryImplementor.class)` and then do something similar to `io.quarkus.hibernate.orm.runtime.schema.SchemaManagementIntegrator#defaultName` on the resulting object:

```
        String name = sf.getName();
        if (name != null) {
            return name;
        }
        Object persistenceUnitName = sf.getProperties().get("hibernate.ejb.persistenceUnitName");
        if (persistenceUnitName != null) {
            return persistenceUnitName.toString();
        }
        return PersistenceUnitUtil.DEFAULT_PERSISTENCE_UNIT_NAME;
```

Not sure if that's much better than your solution, but it *is* consistent with other devmode-related code...

Reply:
changed according to suggestion